src/App.tsx

```diff
@@ -15,36 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'byAlphabet',
+  Length = 'byLength',
+  Default = 'byDefault',
+}
+
+const getPreparedGoods = (
+  goods: string[],
+  sortField?: SortType | null,
+  reverse?: boolean,
+) => {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.Alphabet:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.Length:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | null>(null);
+  const [reverse, setReverse] = useState(false);
+
+  const preparedGoods = getPreparedGoods(goodsFromServer, sortField, reverse);
+  const isReset = sortField || reverse;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SortType.Alphabet,
+          })}
+          onClick={() => setSortField(SortType.Alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button is-success', {
+            'is-light': sortField !== SortType.Length,
+          })}
+          onClick={() => setSortField(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button is-warning', {
+            'is-light': !reverse,
+          })}
+          onClick={() => setReverse(!reverse)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {isReset && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(null);
+              setReverse(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {preparedGoods.map((good: string, index: number) => (
+          <li key={index} data-cy="Good">
```

Comment:
Don't use `index` as a key
https://react.dev/learn/rendering-lists
https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/